src/ssh.c

```diff
@@ -1562,8 +1563,49 @@ int wolfSSH_ReadKey_buffer(const byte* in, word32 inSz, int format,
             ret = WS_SUCCESS;
         }
     }
-    else
+    else if (format == WOLFSSH_FORMAT_PEM) {
+        word32 newKeySz = inSz; /* binary will be smaller than PEM */
+
+        if (*out == NULL) {
+            newKey = (byte*)WMALLOC(newKeySz, heap, DYNTYPE_PRIVKEY);
+            if (newKey == NULL) {
+                return WS_MEMORY_E;
+            }
+            *out = newKey;
+        }
+        else {
+            if (*outSz < newKeySz) {
+                return WS_ERROR;
+            }
+            newKey = *out;
+        }
+
+        /* If it is PEM, convert to ASN1 then process. */
+        ret = wc_KeyPemToDer(in, inSz, newKey, newKeySz, NULL);
+        if (ret > 0) {
+            newKeySz = (word32)ret;
+            ret = WS_SUCCESS;
+        }
+        else {
+            ret = WS_ERROR;
```

Comment:
Perhaps it would be helpful to use/capture or log the actual failure return code from wc_KeyPemToDer

Reply:
Updating.